praas-app-ionic/src/store/user/login.js

```diff
@@ -66,26 +62,25 @@ const localUser = JSON.parse(localStorage.getItem('user'));
 const initialState = localUser
   ? { inflight: false, loggedIn: true, ...localUser }
   : { inflight: false, loggedIn: false };
-export default function login(state = initialState, { type, payload }) {
+export default function login(state = initialState, { type, user, errors }) {
   switch (type) {
     case LOGIN_REQUEST:
       return {
         ...state,
         inflight: true,
-        ...payload.user,
       };
     case LOGIN_SUCCESS:
-      localStorage.setItem('user', JSON.stringify(payload.user));
+      localStorage.setItem('user', JSON.stringify(user));
```

Comment:
Is it required to persist the user in local storage? Why not put it in the store? Anyways when ever we are refreshing, all the api calls including authentication should be made for the fact that the session might get expired in between. I am not sure how it impacts the whole logic of authentication. But this is my first thought

Reply:
This is implemented as in praas-app. we'd need to persist, at least the token. We are using JWT but I dont see any refresh token as part of api response. the tokens should be always needed to be persisted. In a case when the token is expired or revoked, what we would be getting is a 401 which would then be redirected to login page.

Reply:
@m5nv any thoughts on this? I would say all that we have to persist is the access_token here. I am not quite sure if we are following oauth2 here or any other authentication mechanism which provides a refresh token here though. But as you said, if the auth api returns an error, then the user should be kicked out.

Reply:
@vakareeralil as per the `crud-server` we are using JWT. but let me dig deeper to know if we could use a refresh token.
But we can skip it if there is no priority for this.

Reply:
@vakareeralil, @sirajulmcb 

We have not implemented OAuth yet. So there's no notion of refresh token at the moment. It is on the roadmap. To answer your question, we are not following OAuth2 - that's happening in a separate stream.